### ISSUE TITLE
support yaml views inside plugins

### ISSUE THREAD

**Issue by jstrachan** [vmware-tanzu/octant#919]
**Describe the problem/challenge you have**
it would be awesome to be able to reuse the yaml component inside a plugin via `component.NewYAML()`

Currently if you try you get this error:
```
ERROR: api/content_manager.go:130  generate content  {"client-id": "0252e0ec-9052-11ea-bb88-acde48001122", "err": "generate content: unmarshal content response: unmarshal component: unmarshal flexlayout config: unknown view component \"yaml\""
```

**Comment by wwitzel3** [comment]
Would using the Editor in read-only mode work for you? We are planning to phase out the existing YAML component.

https://github.com/vmware-tanzu/octant/blob/master/pkg/view/component/editor.go#L32

It is in the unmarshal switch, so should be available to plugins.

https://github.com/vmware-tanzu/octant/blob/master/pkg/view/component/unmarshal.go#L54

**Comment by jstrachan** [comment]
@wwitzel3 a read only editor view would be awesome - many thanks!

I gave it a go and it seems to render on a FlexLayout as about 2-4 pixels high so not terribly useful for reading the YAML :) - lemme see if I can figure out if its a simple-ish CSS fix

**Comment by GuessWhoSamFoo** [comment]
Looks like https://github.com/vmware-tanzu/octant/blob/master/web/src/app/modules/shared/components/smart/editor/editor.component.scss#L13 can be set to some fixed height like 500px, but I don't know what that line currently does ¯\_(ツ)_/¯

**Comment by bryanl** [comment]
This one is tricky. I set up the editor to use flexbox and it felt like bit twiddling. Another idea is to make it so the editor shows in a modal. Then it would be easier to set its size. Putting it in a large or extra-large modal would probably be the best option.

**Comment by jstrachan** [comment]
@bryanl good point; its about time I figured out how to use Tabs in octant plugins; putting the Editor in a separate tab sounds a nice fix...

**Comment by wwitzel3** [comment]
I believe some work done by @GuessWhoSamFoo and @scothis around the Monaco editor has resolved the sizing issue you were running in to before.
